event/eventQueue/eventQueue.go

```diff
@@ -0,0 +1,132 @@
+package eventqueue
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/go-kit/kit/log"
+	"github.com/xmidt-org/glaukos/event/parsing"
+	"github.com/xmidt-org/webpa-common/basculechecks"
+	"github.com/xmidt-org/webpa-common/logging"
+	"github.com/xmidt-org/webpa-common/semaphore"
+	"github.com/xmidt-org/wrp-go/v3"
+)
+
+var (
+	defaultLogger = log.NewNopLogger()
+)
+
+const (
+	defaultMinMaxWorkers = 5
+	defaultMinQueueSize  = 5
+)
+
+// QueueConfig configures the glaukos queue used to parse incoming events from Caduceus
+type QueueConfig struct {
+	QueueSize  int
+	MaxWorkers int
+}
+
+type EventQueue struct {
+	queue   chan wrp.Message
+	workers semaphore.Interface
+	wg      sync.WaitGroup
+	logger  log.Logger
+	config  QueueConfig
+	parsers parsing.ParsersIn
+	metrics QueueMetricsIn
+}
+
+func NewEventQueue(config QueueConfig, parsers parsing.ParsersIn, metrics QueueMetricsIn, logger log.Logger) (*EventQueue, error) {
+	if parsers.BootTimeParser == nil {
+		return nil, errors.New("No boot time parser")
+	}
+
+	if parsers.MetadataParser == nil {
+		return nil, errors.New("No metadata parser")
+	}
+
+	if config.MaxWorkers < defaultMinMaxWorkers {
+		config.MaxWorkers = defaultMinMaxWorkers
+	}
+
+	if config.QueueSize < defaultMinQueueSize {
+		config.QueueSize = defaultMinQueueSize
+	}
+
+	if logger == nil {
+		logger = defaultLogger
+	}
+
+	queue := make(chan wrp.Message, config.QueueSize)
+	workers := semaphore.New(config.MaxWorkers)
+
+	r := EventQueue{
+		config:  config,
+		queue:   queue,
+		logger:  logger,
+		workers: workers,
+		parsers: parsers,
+		metrics: metrics,
+	}
+
+	return &r, nil
+}
+
+func (e *EventQueue) Start() {
```

Comment:
For starting and stopping we can leverage [uber.fx Lifecycle ](https://pkg.go.dev/go.uber.org/fx#Lifecycle)